### ISSUE TITLE
Use key management from exposure-notifications-server

### ISSUE THREAD

**Issue by ricardobessa1** [google/exposure-notifications-verification-server#133]
### TL;DR
Since key management package of project  **google/exposure-notifications-server** was moved from **internal** to **pkg/keys**, should be useful use this package that have more options instead of **gcpkms** in this project.

I have this change done in a local repo, I can create a pull request with changes if wanted.

### Design

**Proposal**

- Add _KeyManager keys.Config_ on _APIServerConfig_;
- Use the method _keys.KeyManagerFor_ instead of _gcpkms.New_;
- Remove packages _pkg/signer_ and _pkg/gcpkms_;

**Resources**

- [https://github.com/google/exposure-notifications-verification-server/tree/main/pkg/gcpkms](https://github.com/google/exposure-notifications-verification-server/tree/main/pkg/gcpkms)
- [https://github.com/google/exposure-notifications-server/tree/main/pkg/keys](https://github.com/google/exposure-notifications-server/tree/main/pkg/keys)

**Comment by sethvargo** [comment]
I did this as part of https://github.com/google/exposure-notifications-verification-server/pull/121, but we decided to hold off for now as this system architecture is rapidly evolving.

**Comment by ricardobessa1** [comment]
Ok, I understand. But it would be great if we can at least implement other managements like AWS KMS using the KeyManager interface available on **pkg/signer** in this project.

**Comment by sethvargo** [comment]
Hi @ricardobessa1 right now we're focused on GCP for this server.

**Comment by sherifkozman** [comment]
@sethvargo to be clear from a roadmap prespective this means that the env values for AWS KMS at this point are of no use, correct ?

**Comment by sethvargo** [comment]
@sherifkozman I'm not sure what you mean?

**Comment by sherifkozman** [comment]
Docs [here](https://github.com/google/exposure-notifications-server/blob/main/docs/getting-started/deploying.md) mention AWS KMS as a signer option. can this be used as of now or its not implemented as per the discussion above?

**Comment by mikehelmick** [comment]
Hello - 

If the health authority that you are representing needs to run this server on AWS, please reach out directly to Google through your developer relations contact. That is a signal we can use to prioritize work.

Right now, this space is rapidly evolving and we're focusing on remaining functionality before making this operate across clouds.

Because of the architecture of this application, there will always be a GCP dependency (Firebase). Is that acceptable to clients who wish to run the compute on AWS?

@sherifkozman - those docs are for the other server, not the verification server.

**Comment by sherifkozman** [comment]
@mikehelmick my bad as they got mixed up, sorry for that

**Comment by mikehelmick** [comment]
No worries. Please let us know if this is a blocker for a specific health authority - that's something we need to know.

**Comment by ricardobessa1** [comment]
@mikehelmick,  thank you for explanation about the roadmap.
Actually we are only using  _cmd/apiserver_ to handle the mobile reqs and _cmd/adminapi_ as a internal endpoint used by our own frontend. In this scenario we are not using firebase auth implemented on _cmd/server_ and the server is running on AWS with the changes described on this issue to use KMS.

Anyway, I'll close this issue and keep looking the roadmap. Let us know if need any help with any issue or tests with other environments.

**Comment by mikehelmick** [comment]
@sherifkozman @ricardobessa1  - can you email us 
address listed here -> https://github.com/google/exposure-notifications-server#issues-and-questions

I would like to know which health authorities you're working with and what your proposed trial / launch date(s) are. That can inform priorities for us.

Re KMS in particular, and the roadmap. As it relates to multi-tenancy, this server may need to provision and rotate the signing keys and that's a broader surface area to support multiple KMS implementations on. We want to finish out that design work before we expand supported services.

**Comment by sherifkozman** [comment]
@mikehelmick i believe we are already on email loops about the HAs we are working with through PathCheck Foundation, i will make sure you are tagged if you are not there.

**Comment by mikehelmick** [comment]
to have this work w/ signing key management, https://github.com/google/exposure-notifications-server/issues/916 will need to be done on the ENS repo.
